File: leetcode/algorithms/array/teemo-attacking.java
/*
	leetcode
	category: "Algorithms"
	problem title: "Teemo Attacking"
*/


class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int totalPoisonTime = 0;
        int low=0,high=0;
        boolean isOverlap=false;
        for(int i=0;i<timeSeries.length;++i){
            int t=timeSeries[i]-1;
            if(!isOverlap){
                low=t;    
            }
            high=t+duration;
            if(i!=timeSeries.length-1 && timeSeries[i+1]-1<high){
                isOverlap=true;
            } else {
                isOverlap=false;
            }
            if(!isOverlap){
                totalPoisonTime+=(high-low);
            }
        }
        return totalPoisonTime;
    }
}

File: leetcode/algorithms/array/teemo-attacking.ts
/*
	leetcode
	category: "Algorithms"
	problem title: "Teemo Attacking"
*/

function findPoisonedDuration(timeSeries: number[], duration: number): number {
    
    let poisonedTill: number = -1,totalDuration: number = 0;
    for(let attackTime of timeSeries){
        totalDuration+=duration-Math.max(0,poisonedTill-attackTime);
        poisonedTill=attackTime+duration;
    }
    
    return totalDuration;

};